repose-aggregator/docs/build.gradle

```diff
@@ -0,0 +1,286 @@
+import org.ajoberstar.grgit.Grgit
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.regex.Pattern
+
+apply plugin: 'org.openrepose.gradle.plugins.linkchecker'
+
+def username = System.properties['user.name']
+def uid = ["id", "-u", username].execute().text.trim()
+def gid = ["id", "-g", username].execute().text.trim()
+
+task asciibinder(group: 'build', description: 'Builds the latest version of the docs') {
+    inputs.dir(file("${project.projectDir}/src"))
+    outputs.dir(file("${project.buildDir}/asciibinder"))
+    doLast {
+        copy {
+            from(fileTree("${project.projectDir}/src/asciibinder"))
+            into("${project.buildDir}/asciibinder")
+        }
+        copy {
+            from(fileTree("${project.projectDir}/src/templates"))
+            into("${project.buildDir}/asciibinder")
+            expand(version: version)
+        }
+
+        def grgit = Grgit.init(dir: "${project.buildDir}/asciibinder")
+        grgit.add(patterns: ['.'])
+        grgit.commit(message: 'initial_commit')
+
+        def buildProcess =
+            "docker build --tag local/ascii_binder ${project.projectDir}/src/docker/AsciiBinder/"
+                .execute(null, file("${project.buildDir}/asciibinder"))
+        buildProcess.waitFor()
+        if (buildProcess.exitValue() != 0) {
+            logger.error new String(buildProcess.getInputStream().bytes)
+            logger.error new String(buildProcess.getErrorStream().bytes)
+        } else {
+            logger.info new String(buildProcess.getInputStream().bytes)
+        }
+
+        def runProcess =
+            "docker run --tty --user ${uid}:${gid} --volume ${project.buildDir}/asciibinder:/docs:z --volume ${project.rootDir}/repose-aggregator:/repose-aggregator:z local/ascii_binder"
+                .execute(null, file("${project.buildDir}/asciibinder"))
+        try {
+            runProcess.waitFor()
+            def runProcessInputStreamAsString = new String(runProcess.getInputStream().bytes)
+            if (runProcessInputStreamAsString =~ /asciidoctor: WARNING:/ ||
+                runProcessInputStreamAsString =~ /asciidoctor: ERROR:/ ||
+                runProcess.exitValue() != 0) {
+                logger.error(runProcessInputStreamAsString)
+                logger.error(new String(runProcess.getErrorStream().bytes))
+                throw new RuntimeException("See output above for specific cause of AsciiBinder failure(s).")
+            } else {
+                logger.info(runProcessInputStreamAsString)
+                copy {
+                    from(fileTree("${project.projectDir}/src/redirects/index.html"))
+                    into("${project.buildDir}/asciibinder/_package/main/${version}/")
+                    expand(target: "./welcome/index.html")
+                }
+            }
+        } finally {
+            file("${project.buildDir}/asciibinder/.git").deleteDir()
```

Comment:
I'm fine with that, but in that case, I would suggest just deleting the `.git` directory before we `init` a new one. That way we don't have to use a try/finally block, we do handle the case where an errant `.git` directory appears, and the Git repository is left in place after the build in case we want or need to inspect it for some reason.

Reply:
Sure. Updated.